src/utils.js

```diff
@@ -33,36 +38,42 @@ export function clone(target) {
  * Merge any number of objects into a provided object.
  * @private
  */
-export function merge() {
-  let copy = null
-  let subject = null
+export function merge(): Object {
+  var copy = {}
+  var first = copy
+  var dirty = false
 
-  for (var i = 0, len = arguments.length; i < len; i++) {
-    copy = copy || arguments[i]
-    subject = subject || copy
+  for (var i = arguments.length - 1; i >= 0; i--) {
```

Comment:
why do you use `arguments` vs `...args` or something (I might be getting that wrong). Just curious, don't think you need a change or anything.